spring-cloud-netflix-core/src/main/java/org/springframework/cloud/netflix/ribbon/RibbonClientHttpRequestFactory.java

```diff
@@ -48,6 +48,9 @@ public RibbonClientHttpRequestFactory(SpringClientFactory clientFactory, LoadBal
 	public ClientHttpRequest createRequest(URI originalUri, HttpMethod httpMethod)
 			throws IOException {
         String serviceId = originalUri.getHost();
+        if (serviceId == null) {
+            throw new IOException("Invalid hostname in the URI [" + originalUri.toASCIIString() + "]");
```

Comment:
If this is a real problem we should do some validation much earlier, during startup rather than runtime.